### PR TITLE
K8s guidance: running a single container

### PR DESCRIPTION
**Pull request by ElijahCaine**
First shot at writing a k8s doc.  Covers basic use-cases like starting pods, services, and management with `json`/`yaml`.

Feedback welcome.
refs coreos-inc/coreos-pages/issues/583
cc: @joshix @philips et al.

### CODE REVIEW

File: kubernetes/running-single-container.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+First you need to define a deployment. This can be done a few ways, the fastest is with `kubectl run $DEPLOYMENT_NAME --image=$IMAGE --port=$PORT --replicas=$NUM_REPLICAS`. For example:
+
+```
+$ kubectl run webserver --image=nginx --port=80 --replicas=1
+deployment "webserver" created
+$ kubect get deployments
+NAME        DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+webserver   1         1         1            1           33s
+$ kubect get pods
```
Comment (robszumski) [inline kubernetes/running-single-container.md]: Might be a good idea to split out the pods from this code block, so we can add a sentence in front of it.

> ```
$ kubectl run webserver --image=nginx --port=80 --replicas=1
deployment "webserver" created
$ kubect get deployments
NAME        DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
webserver   1         1         1            1           33s
```

> As you can see, our deployment started a single copy of the container. To view each container instance, called a pod, run:

> ```
$ kubect get pods
NAME                         READY     STATUS    RESTARTS   AGE
webserver-3274421635-a4al1   1/1       Running   0          35m

File: kubernetes/running-single-container.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+webserver-3274421635-a4al1   1/1       Running   0          35m
+```
+
+## Start a service
+
+This is a great start, but we can't access our container yet. To do that we need to define a Kubernetes service, which we will do with `kubernetes expose deployment $DEPLOYMENT_NAME --type=$EXPOSE_TYPE`.
```
Comment (robszumski) [inline kubernetes/running-single-container.md]: same thing here with skipping the variables

File: kubernetes/running-single-container.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+$ kubectl get pods
+NAME                         READY     STATUS    RESTARTS   AGE
+webserver-3274421635-7zupw   1/1       Running   0          53s
+webserver-3274421635-a19vl   1/1       Running   0          53s
+webserver-3274421635-a4al1   1/1       Running   0          18m
+webserver-3274421635-isrmc   1/1       Running   0          53s
+webserver-3274421635-wwnum   1/1       Running   0          53s
+```
+
+Replicas can be increased or decreased based on you see fit and pods will be created and deleted accordingly.
```
Comment (robszumski) [inline kubernetes/running-single-container.md]: might be good to cover the rolling update stuff there. "now if you updated the image..."

File: kubernetes/running-single-container.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+[comment]: # (### AWS)
+
+[comment]: # (### Tectonic)
+
+[comment]: # (### GCE)
+
+## Scale the service
+
+Let's say you spoke too soon when you said you 'Only need one container'. We can scale a given deployment with `kubectl scale $DEPLOYMENT_NAME --replcias=$NUM`
```
Comment (robszumski) [inline kubernetes/running-single-container.md]: consider skipping variables

File: kubernetes/running-single-container.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+You can save a json or yaml file describing a running application with the following command:
+
+```
+$ kubectl get deployments,services webserver -o yaml > webserver.yml
+```
+
+Feel free to read through `webserver.yml`, but for our purposes it's not necessary.
+
+To destroy the deployments and services in this file pass the `-f` flag to `kubectl delete`.
```
Comment (robszumski) [inline kubernetes/running-single-container.md]: I'd explain this flag once for the entire doc. "you can use -f to reference a file..."

File: kubernetes/running-single-container.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
+# Running a single container
+
+[comment]: # (TODO: Environment Setup Guide / Reference?)
+
+Let's say you just want to start one *one* container with Kubernetes. How would you do that?
+
+## Start a deployment
+
+First you need to define a deployment. This can be done a few ways, the fastest is with `kubectl run $DEPLOYMENT_NAME --image=$IMAGE --port=$PORT --replicas=$NUM_REPLICAS`. For example:
```
Comment (ElijahCaine) [inline kubernetes/running-single-container.md]: @robszumski I know from experience the structure of the k8s commands can be confusing so I figured adding a template might help.  If it really clutters the document we can get rid of it before merge, but for now I'd like to keep it.
I added everything else in this thread.

**Comment by ElijahCaine** [comment]
@robszumski Thank's for the feedback. I didn't delete the `$VARIABLE` commands. I do think they can help beginners better understand the commands they're running in more abstractly (speaking as somebody that *just* learned a lot of this).  I am open to getting rid of them if you / others feel strongly that they don't help though.
As a compromise we can link to a relevant upstream doc for each section.

**Comment by joshix** [comment]
Closing for age; used in pieces in tutorials and other locations.

**Pull request closed without merging**